test/e2e/query_frontend_test.go

```diff
@@ -43,12 +45,44 @@ func TestQueryFrontend(t *testing.T) {
 
 	now := time.Now()
 
-	prom, sidecar := e2ethanos.NewPrometheusWithSidecar(e, "1", e2ethanos.DefaultPromConfig("test", 0, "", "", e2ethanos.LocalPrometheusTarget), "", e2ethanos.DefaultPrometheusImage(), "")
-	testutil.Ok(t, e2e.StartAndWaitReady(prom, sidecar))
+	i := e2ethanos.NewReceiveBuilder(e, "ingestor").WithIngestionEnabled().Init()
+	testutil.Ok(t, e2e.StartAndWaitReady(i))
+	
+	// prom, sidecar := e2ethanos.NewPrometheusWithSidecar(e, "1", e2ethanos.DefaultPromConfig("test", 0, "", "", e2ethanos.LocalPrometheusTarget), "", e2ethanos.DefaultPrometheusImage(), "")
+	// testutil.Ok(t, e2e.StartAndWaitReady(prom, sidecar))
 
-	q := e2ethanos.NewQuerierBuilder(e, "1", sidecar.InternalEndpoint("grpc")).Init()
+	// prom := e2ethanos.NewPrometheus(e, "1", e2ethanos.DefaultPromConfig("test", 0, e2ethanos.RemoteWriteEndpoint(i.InternalEndpoint("remote-write")), "", e2ethanos.LocalPrometheusTarget), "", e2ethanos.DefaultPrometheusImage())
+	// testutil.Ok(t, e2e.StartAndWaitReady(prom))
+
+	q := e2ethanos.NewQuerierBuilder(e, "1", i.InternalEndpoint("grpc")).Init()
 	testutil.Ok(t, e2e.StartAndWaitReady(q))
 
+	tsMillis := timestamp.FromTime(now)
+	var seriest []prompb.TimeSeries
+	metric1 := []prompb.Label{
+		{Name: "A", Value: "K6sjsNNczPl"},
+		{Name: labels.MetricName, Value: "fingerprint_collision"},
+	}
+	metric2 := []prompb.Label{
+		{Name: "A", Value: "cswpLMIZpwt"},
+		{Name: labels.MetricName, Value: "fingerprint_collision"},
+	}
```

Comment:
Why choosing those labels? The series labels need to match what we have in the query results below.

Reply:
ok so to synthesize new fake metrics, I would have to define another function similar to synthesizeFakeMetricSamples in query_test for the receiver, right?

Reply:
Testing this in another branch for now, will update the final changes in this branch.

Reply:
I don't think you should include `receive`, `replica`, `tenant_id` as part of the input labels. They should be attached by Receiver instance itself

Reply:
replica label is needed (giving an error otherwise), I think that was attached by the Prometheus instance